src/depth_limit.rs

```diff
@@ -82,6 +87,15 @@ impl<T: Decode> DecodeLimit for T {
 		}
 	}
 
+	fn decode_and_advance_with_depth_limit<I: Input>(limit: u32, input: &mut I) -> Result<Self, Error> {
+		let mut input = DepthTrackingInput {
+			input,
+			depth: 0,
+			max_depth: limit,
+		};
+		T::decode(&mut input)
+	}
+
 	fn decode_with_depth_limit(limit: u32, input: &[u8]) -> Result<Self, Error> {
```

Comment:
I think we need to put a deprecated on this

Reply:
Not really. However, we should create an issue to merge both of these methods in the next major release. @KiChjang please also create some new milestone for the next major and attach the issue.

Reply:
Filed as #287 and added a SCALE codec 3.0 milestone.